internal/common/model.go

```diff
@@ -1,17 +1,43 @@
 package common
 
-import "github.com/razorpay/metro/internal/app"
+import (
+	"fmt"
+
+	"github.com/razorpay/metro/internal/app"
+)
 
 // IModel interface which all models should implement
 type IModel interface {
 	// Key returns the key to store the model against
 	Key() string
 	// Prefix returns the key prefix used by the module
 	Prefix() string
+	// Returns the version of the saved model in the registry. Version changes for every update.
+	GetVersionID() (string, error)
+	//Sets the version of the saved model from the registry. Never to be used by business logic.
+	SetVersionID(vid string)
 }
 
 // BaseModel implements basic model functionality
-type BaseModel struct{}
+type BaseModel struct {
+	versionID    string
+	isVersionSet bool
+}
+
+//SetVersionID Base implementation of set version id
+func (b *BaseModel) SetVersionID(vid string) {
+	b.isVersionSet = true
+	b.versionID = vid
+}
+
+//GetVersionID Base implementation of get version id. errors out if called without setting the version id.
+func (b *BaseModel) GetVersionID() (string, error) {
+	if !b.isVersionSet {
```

Comment:
can we just do a nil check on version attribute instead ?

Reply:
As discussed, lets do a compat test else we can assume current version to be 0 in all entities

Reply:
Compat test done. Subscriptions are scheduled correctly. Have set default version as "0"